assignments/arrays.js

```diff
@@ -62,33 +62,79 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
-
-// ==== Challenge 2 ====
-// The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
-
-// ==== Challenge 3 ====
-// The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
-
-// ==== Challenge 4 ====
-// The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
-
-// ==== Challenge 5 ====
-// The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars = [];
-console.log(); 
-
-// ==== Challenge 6 ====
-// A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
-let BMWAndAudi = [];
-console.log();
-
+// The dealer can't recall the information for a car with an
+// id of 33 on his lot. Help the dealer find out which car has an id of 33 by 
+//logging the car's year, make, and model in the console log provided to you below:
 
+function carCheck(obj,id){
+    for ( let i = 0 ; i < obj.length ;i++){
+        if(obj[i].id === id) {
+            console.log(`Car ${id} is a ${obj[i].car_year} ${obj[i].car_make} ${obj[i].car_model}` );
+            }
+        }
+    
+    }
+    
+    carCheck(inventory,33);
+    console.log("===========================================================")
+    
+    // ==== Challenge 2 ====
+    // The dealer needs the information on the last car in their inventory.  
+    //What is the make and model of the last car in the inventory?  Log the make and model into the console.
+    
+    carCheck(inventory,inventory.length);
+    // ==== Challenge 3 ====
+    // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into
+    // alphabetical order and log the results in the console
+    
+    let carModels = [];
+    for (let i = 0 ; i < inventory.length ;i++){
+    carModels.push(inventory[i].car_make);
+    }
+    carModels = carModels.sort();
+    console.log("===========================================================")
+    console.log("Car models are ");
+    for ( i in carModels){
+        console.log(carModels[i]);
+    }
```

Comment:
your for loop is perfectly constructed to accomplish this task, but you're actually logging the manufacturers (car_make) rather than the model (car_model) of the cars, whoops.